### PR TITLE
Dockerize

### PR DESCRIPTION
**Pull request by bhgottfried**
Containerize our stack. The entire project can be built on a system with docker and docker-compose via `docker-compose up --build`. This involved substantial refactoring across the project, but will hopefully not break any other branches during the merge.

### CODE REVIEW

File: frontend/src/Review.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ class ReviewContainer extends React.Component {
     }
 
     sendData = (rest_id, review) => {
-        fetch("/submit_review", {
+        fetch("/cravr/submit_review", {
```
Comment (EliasTalcott) [inline frontend/src/Review.js:45]: I see another fetch("<function in app.py>") on line 12 of this file. Should that be modified in this way as well?

**Review by EliasTalcott — APPROVED** [review]
These changes seem good as far as I can tell/test. Dockerfile and Docker Compose look consistent with what I learned the other day. One comment on routing in Review.js, but everything else checks out.

I also ran the backend and frontend together locally and they are not broken by these changes.

File: frontend/src/Review.js (hunk 1 above)
Comment (bhgottfried) [reply frontend/src/Review.js:45]: Yes, youre very correct. I missed that when I did a full directory search for fetch somehow...

**Review by jhyndma — APPROVED** [review]
Changes look great. Dockerizing did not break anything in the front and back ends. Approved

**Pull request merged**